ui/app/AppLayouts/Chat/controls/UserDelegate.qml

```diff
@@ -9,138 +9,95 @@ import StatusQ.Components 0.1
-                    if (statusType === Constants.statusType_DoNotDisturb) {
-                        return Style.current.red;
-                    } else if (isCurrentUser || (lastSeenMinutesAgo < 5.5)) {
-                        return Style.current.green;
-                    } else if (((statusType !== -1) && (lastSeenMinutesAgo > 5.5)) ||
-                               ((statusType === -1) && (lastSeenMinutesAgo < 7))) {
-                        return Style.current.orange;
-                    } else if ((statusType === -1) && (lastSeenMinutesAgo > 7)) {
-                        return "transparent";
-                    }
-                } else {
-                    return "transparent";
-                }
+    Connections {
+        enabled: !!root.contactsList
+        target: root.contactsList
+        onContactChanged: {
+            if (pubkey === root.publicKey) {
+                root.profileImage = !!appMain.getProfileImage(root.publicKey) ?
+                            appMain.getProfileImage(root.publicKey) : ""
             }
         }
+    }
 
-        MouseArea {
-            enabled: enableMouseArea
-            cursorShape: enabled ? Qt.PointingHandCursor : Qt.ArrowCursor
-            acceptedButtons: Qt.LeftButton | Qt.RightButton
-            anchors.fill: parent
-            hoverEnabled: true
-            onEntered: {
-                wrapper.hovered = true
-            }
-            onExited: {
-                wrapper.hovered = false
-            }
-            onClicked: {
-                if (mouse.button === Qt.LeftButton) {
-                    //TODO remove dynamic scoping
-                    openProfilePopup(wrapper.name, wrapper.publicKey, (wrapper.profileImage || wrapper.identicon), "", appMain.getUserNickname(wrapper.publicKey));
-                }
-                 else if (mouse.button === Qt.RightButton && !!messageContextMenu) {
-                    // Set parent, X & Y positions for the messageContextMenu
-                    messageContextMenu.parent = rectangle
-                    messageContextMenu.setXPosition = function() { return 0}
-                    messageContextMenu.setYPosition = function() { return rectangle.height}
-                    messageContextMenu.isProfile = true;
-                    messageContextMenu.show(wrapper.name, wrapper.publicKey, (wrapper.profileImage || wrapper.identicon), "", appMain.getUserNickname(wrapper.publicKey))
+    StatusBadge {
+        id: statusBadge
+        width: 15
+        height: 15
+        anchors.left: parent.left
+        anchors.leftMargin: 22
+        anchors.bottom: parent.bottom
+        anchors.bottomMargin: 6
+        visible: root.isOnline && !((root.statusType === -1) && (lastSeenMinutesAgo > 7))
+        border.width: 3
+        border.color: root.sensor.containsMouse ? Theme.palette.baseColor2 : Theme.palette.baseColor4
+        property real lastSeenMinutesAgo: ((currentTime/1000 - parseInt(lastSeen)) / 60);
+        color: {
```

Comment:
just a suggestion maybe to move this to the store? the logic of calculating the color of the status indicator?

Reply:
That's a good suggestion. We could even go as far as adding an enum to this component and the variant needs to be set at the call side (using that store API)

Reply:
aaah yes, but then we will again be putting if cases against the ENUM in the qml? 
Also shouldn't this simply come from backend? at least the logic of calculating the various states?